uspace/app/layout/layout.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2019 Matthieu Riolo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * - Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * - The name of the author may not be used to endorse or promote products
+ *   derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** @addtogroup layout
+ * @{
+ */
+/**
+ * @file
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <str.h>
+#include <str_error.h>
+#include <ipc/services.h>
+#include <ipc/input.h>
+#include <io/input.h>
+#include <abi/ipc/interfaces.h>
+#include <loc.h>
+
+static const char *cmdname = "layout";
+
+/* Dispays help for layout in various levels */
+static void print_help()
+{
+	printf("Changes, lists or displays the current keyboard layout.\n");
+
+	printf(
+	    "Usage: %s\n"
+	    "\t%s list             lists all layouts\n"
+	    "\t%s get              displays currently set layout\n"
+	    "\t%s set <layout>     changes to the new layout\n",
+	    cmdname, cmdname, cmdname, cmdname);
+}
+
+static async_sess_t *hid_exchange_start()
```

Comment:
Why don't you use input_open() to get an input_t object and use that to talk to the input server?

Reply:
Also, print_help() is not valid syntax in ANSI/ISO C.

Reply:
> Why don't you use input_open() to get an input_t object and use that to talk to the input server?

I remembering looking into that function and coming to the conclusion that this is an overkill (considering the refactoring it needs). The function is needed for a long time session while I need a single call.


> print_help() is not valid syntax in ANSI/ISO C

I'm sorry I cannot follow. What's wrong with the naming?

Reply:
> > print_help() is not valid syntax in ANSI/ISO C
> 
> I'm sorry I cannot follow. What's wrong with the naming?

I think what he means is that a function with no arguments is declared as `print_help(void)`. Empty parentheses mean undeclared list of arguments (or declared elsewhere), and is not accepted in modern standard C, although GNU still accepts the syntax as an extension.